parsers/jscript.c

```diff
@@ -592,6 +593,17 @@ static int makeFunctionTag (tokenInfo *const token, vString *const signature, bo
 	return makeFunctionTagCommon (token, signature, generator, false);
 }
 
+static bool isClassName (tokenInfo *const name)
+{
+	char * p = strrchr(vStringValue (name->string), '.');
+	if (p == NULL)
+		p = vStringValue (name->string);
+	else
+		p++;
+
+	return isupper(*p);
```

Comment:
In my understanding, a cast operator `(unsigned int)` is needed in the front of `*p`.
isupper() expects int but only -1 is the acceptable negative value; the behavior of isupper() is not defined in the standard C spec.
On the other hand, we cannot have an assume whether char is `unsigned char` or `singed char`. So `*p` can be a negative value other than -1.

See https://github.com/universal-ctags/ctags/pull/1697 .

Reply:
Good point.  Fixed.